build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -0,0 +1,267 @@
+            {
+                string constantsYamlSubPath = Path.Combine("build", Constants.ConstantsYaml);
+                List<ConstantsYamlFile> yamlConstantsObjs = await this.yamlFileService.ReadConstantsYamlFileAsync(constantsYamlSubPath);
+                this.UpdateOryxConstantsForNewVersions(newDotNetVersions, yamlConstantsObjs);
+            }
+        }
+
+        public async Task<List<DotNetVersion>> GetNewDotNetVersionsAsync()
+        {
+            List<DotNetVersion> dotNetVersions = new List<DotNetVersion>();
+
+            // Deserialize release metadata
+            var response = await this.httpService.GetDataAsync(DotNetConstants.ReleasesIndexJsonUrl);
+            var releaseNotes = response == null ? null : JsonConvert.DeserializeObject<ReleaseNotes>(response);
+            var releasesIndex = releaseNotes == null ? new List<ReleaseNote>() : releaseNotes.ReleaseIndexes;
+            foreach (var releaseIndex in releasesIndex)
+            {
+                // If the latest release version is not within the acceptable range (inclusive),
+                // or if the sdk version is present in the set of ORYX SDK versions,
+                // skip to the next version in the loop.
+                if (this.ReleaseVersionIsNotInRangeOrSdkVersionAlreadyExists(
+                    releaseIndex.LatestRelease, releaseIndex.LatestSdk))
+                {
+                    continue;
+                }
+
+                // Get the actual release information from releases.json
+                string releasesJsonUrl = releaseIndex.ReleasesJsonUrl;
+                response = await this.httpService.GetDataAsync(releasesJsonUrl);
+                var releasesJson = JsonConvert.DeserializeObject<ReleasesJson>(response);
+                var releases = releasesJson == null ? new List<Release>() : releasesJson.Releases;
+                foreach (var release in releases)
+                {
+                    // Check again since the "release" is separate from "releaseIndex".
+                    if (this.ReleaseVersionIsNotInRangeOrSdkVersionAlreadyExists(
+                        release.ReleaseVersion, release.Sdk.Version))
+                    {
+                        continue;
+                    }
+
+                    var dotNetSdkVersion = this.GetDotNetVersion(release, DotNetConstants.SdkName);
+                    dotNetVersions.Add(dotNetSdkVersion);
+
+                    // Runtime (netcore)
+                    var dotNetCoreVersion = this.GetDotNetVersion(release, DotNetConstants.DotNetCoreName);
+                    dotNetVersions.Add(dotNetCoreVersion);
+
+                    // Runtime (aspnetcore)
+                    var dotNetAspCoreVersion = this.GetDotNetVersion(release, DotNetConstants.DotNetAspCoreName);
+                    dotNetVersions.Add(dotNetAspCoreVersion);
+                }
+            }
+
+            return dotNetVersions = dotNetVersions.OrderBy(v => v.Version).ToList();
```

Comment:
Nit: please consider removing the re-assignment on `dotNetVersions` as we're about to return anyway.

Reply:
Thanks for catching this one!